frontend/src/features/properties/map/research/update/UpdateResearchContainer.tsx

```diff
@@ -0,0 +1,115 @@
+import MapSideBarLayout from 'features/mapSideBar/layout/MapSideBarLayout';
+import { Formik, FormikProps } from 'formik';
+import { Api_ResearchFile } from 'models/api/ResearchFile';
+import * as React from 'react';
+import { useEffect, useRef, useState } from 'react';
+import { MdTopic } from 'react-icons/md';
+import { Prompt } from 'react-router-dom';
+import styled from 'styled-components';
+
+import ResearchFooter from '../common/ResearchFooter';
+import { useGetResearch } from '../hooks/useGetResearch';
+import { useUpdateResearch } from '../hooks/useUpdateResearch';
+import { UpdateResearchFormModel } from './models';
+import UpdateResearchForm from './UpdateResearchForm';
+import UpdateResearchHeader from './UpdateResearchHeader';
+
+export interface IUpdateResearchContainerProps {
+  researchFileId: number;
+  onClose: () => void;
+}
+
+export const UpdateResearchContainer: React.FunctionComponent<IUpdateResearchContainerProps> = props => {
+  const formikRef = useRef<FormikProps<UpdateResearchFormModel>>(null);
+  const { retrieveResearchFile } = useGetResearch();
+  const { updateResearchFile } = useUpdateResearch();
+
+  const [researchFile, setResearchFile] = useState<Api_ResearchFile | undefined>(undefined);
+  const [initialForm, setForm] = useState<UpdateResearchFormModel | undefined>(undefined);
+
+  useEffect(() => {
+    async function fetchResearchFile() {
+      var retrieved = await retrieveResearchFile(props.researchFileId);
+      setResearchFile(retrieved);
+      if (retrieved !== undefined) {
+        setForm(UpdateResearchFormModel.fromApi(retrieved));
```

Comment:
Not seeing loading logic on this screen, is there a reason for that?

Reply:
I moved the loading logic to a simple return of the component.